File: repos_estudiantes_ALVAREZ_primera_entrega/CARDON_PAZ_LUIS_ABDEL/INTRODUCCION_POO/EJERCICIO_5/Persona.java
package INTRODUCCION_POO.EJERCICIO_5;

public class Persona {
    private String nombre;
    private String materno;
    private String paterno;
    private int edad;
    private int ci;

    public Persona(String nombre, String paterno, String materno, int edad, int ci){
        this.nombre = nombre;
        this.materno = materno;
        this.paterno = paterno;
        this.edad = edad;
        this.ci = ci;
    }

    public void mostrarDatos(){
        System.out.println("Nombre: " + this.nombre + " " + this.paterno + " " + this.materno + ", Edad: " + this.edad + ", CI: " + this.ci);

    }
    
    public void mayorDeEdad(){
        if (this.edad >= 18){
            System.out.println("es mayor de edad");

        }else{
            System.out.println("es menor de edad");
        }
    }

    public String getPaterno(){
        return this.paterno;
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_PACHECO_BRYAN_ORLANDO/POO/Ejer5/Persona.py
class Persona:
    # a) 
    def __init__(self,nombre,paterno,materno,edad,ci):
        self.nombre=nombre
        self.paterno=paterno
        self.materno=materno
        self.edad=edad
        self.ci=ci

    # b) 
    def mostrarDatos(self):
        print("Nombre:",self.nombre,self.paterno,self.materno)
        print("Edad:",self.edad,"CI:",self.ci)

    # c)
    def mayorDeEdad(self):
        return self.edad>=18

    # d)
    def modificarEdad(self,nuevo):
        self.edad=nuevo

    # e) 
    def mismoApellido(self,otra):
        return self.paterno==otra.paterno

# a) Instanciar
p1=Persona("Bryan","Apaza","Lopez",18,"123456")
p2=Persona("Carlos","Apaza","Perez",16,"654321")

p1.mostrarDatos()
p2.mostrarDatos()

print("P1 mayor de edad?",p1.mayorDeEdad())
print("P2 mayor de edad?",p2.mayorDeEdad())

p2.modificarEdad(20)
print("Edad modificada P2:",p2.edad)

print("Mismo apellido paterno?",p1.mismoApellido(p2))
